platform-sdk/docs/proposals/001-merkle-state-api/001-merkle-state-api.md

```diff
@@ -0,0 +1,65 @@
+# Merkle State API
+
+## Summary
+
+Refactor the Merkle tree so that it can be fully represented by the State API objects.
+Create a module that will provide a simple and easy-to-use API for the Merkle tree.
+
+## Purpose and Context
+
+### Why?
+
+The Merkle tree is a rather complex data structure, and to make interaction easier, we need to hide this complexity behind the API.
+Another reason to have this API is to provide additional flexibility in changing the implementation details of the Merkle tree.
+As long as they are hidden, we are free to change them.
+
+Note that this API is already implemented as a part of the Service Modularization project, but these classes will be moved
+to a separate module as a part of this proposal.
+
+### Context
+
+There are two issues that need to be addressed:
+
+* We need to have a public API for the Merkle tree that is simple and easy to use, residing in a separate module with a minimal set of dependencies.
+* The Merkle tree has to be fully expressed in terms of State API. Currently, `PlatformState` stands out and needs to be refactored.[](https://)
+
+## Requirements
+
+- `PlatformState` should be refactored to a singleton object as defined by the State API.
+- there should be a new module - `swirlds-state-api` - to host a set of interfaces, records, and abstract classes that represent the Hedera state.
+  This module should have a minimal set of dependencies. The Block Node should not have a compile-time dependency
+  on any other modules but this one to interact with the state.
+- there should be a new module - `swirlds-state-impl` module to host the implementation of these interfaces and helper classes.
```

Comment:
We don't plan any other implementation. My initial thinking was that API and implementation in one module will be confusing, and I followed the pattern that we have for other modules - `swirlds-config-api` / `swirlds-config-impl` or `swirlds-metrics-api`/ `swirlds-metrics-impl`. However, I don't have a strong opinion about this and, in fact, that would be easier for me to implement as well, if we didn't have to have another module.

Reply:
For metrics and config we did it because it must work with a different config / metrics system. If you want to use then platform in a spring boot application you want to have the spring metrics system be the implementation of our APIs. That was a wish since the beginning. If we do not assume to have a second implementation than I would put both in 1 module to minimize the module count.

Reply:
okay, fair enough. I'll remove the notion of `swirlds-state-impl`